scripts/ci/check_compliance.py

```diff
@@ -262,6 +263,38 @@ def run(self):
         for file in path.glob("**/board.yml"):
             self.check_board_file(file, vendor_prefixes)
 
+
+class ClangFormatCheck(ComplianceTest):
+    """
+    Check if clang-format reports any issues
+    """
+    name = "ClangFormat"
+    doc = "See https://docs.zephyrproject.org/latest/contribute/guidelines.html#clang-format for more details."
+    path_hint = "<git-top>"
+
+    def run(self):
+        for file in get_files():
+            diff = subprocess.Popen(('git', 'diff', '-U0', '--no-color', COMMIT_RANGE, '--', file),
+                                    stdout=subprocess.PIPE,
+                                    cwd=GIT_TOP)
+            try:
+                subprocess.run(('clang-format-diff.py', '-p1'),
+                               check=True,
+                               stdin=diff.stdout,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT,
+                               cwd=GIT_TOP)
+
+            except subprocess.CalledProcessError as ex:
+                patchset = unidiff.PatchSet.from_string(ex.output, encoding="utf-8")
+                for patch in patchset:
+                    for hunk in patch:
+                        # Set the annotation to the last line of the hunk
+                        line = hunk.source_start + hunk.source_length - 1
```

Comment:
the position (line #) of the notice looks weird in the test PR you shared? I think this should be adjusted as it's awkward to get the notice placed at a line that, in this particular example, is not even problematic.

Reply:
It's on the end of the hunk, which I think makes sense? Where would you put it otherwise?

The image in my previous comment is of by one as this was the previous version, the test PR looks good IMO.

Reply:
Annotations can be set to more lines using `endLine`, will update.

Reply:
> Annotations can be set to more lines using `endLine`, will update.

ya, that'd help, I think. I think my main concern is that if it's a *line* annotation the actual position should somewhat make sense and directly point at something actionable

Reply:
Update, the start line is now printed in the workflow, but the annotation does appear to be the same as before.